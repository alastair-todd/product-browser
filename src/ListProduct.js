// ListProduct.js
import React from 'react'
import ProductImageContainer from './ProductImageContainer'
import {Button} from 'react-bootstrap'
import {LinkContainer} from 'react-router-bootstrap'

class ListProduct extends React.Component {
  render () {
    let product = this.props.product
    return (
      <div className="product-wrapper">
        <h2>{product.name}</h2>
        <ProductImageContainer product={product} />
        <p>{product.description}</p>
        <LinkContainer to={{ pathname: `/product/${product.id}` }}>
          <Button>View More</Button>
        </LinkContainer>
      </div>
      )
  }
  viewMore () {
    console.log('go product')
  }
}
ListProduct.propTypes = {
  product: React.PropTypes.object.isRequired
}
export default ListProduct
