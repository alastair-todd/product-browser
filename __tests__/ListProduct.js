// ListProduct.react-test.js
import React from 'react'
import ListProduct from '../src/ListProduct'
import renderer from 'react-test-renderer'

jest.mock('react-dom')

describe('ListProduct', () => {
  it('is rendered', () => {
    let product = {
      id: '5076c659-d921-419a-b0e5-6e20c8320721',
      name: 'test product',
      description: 'test product description',
      amount: 450
    }
    const component = renderer.create(
      <ListProduct product={product} />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
