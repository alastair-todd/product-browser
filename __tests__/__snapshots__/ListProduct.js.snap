exports[`ListProduct is rendered 1`] = `
<div
  className="product-wrapper">
  <h2>
    test product
  </h2>
  <div
    className="no-image" />
  <p>
    test product description
  </p>
  <div>
    <button
      action="push"
      className="btn btn-default"
      disabled={false}
      onClick={[Function anonymous]}
      type="button">
      View More
    </button>
  </div>
</div>
`;
